### PR TITLE
fix matmul

### PR DESCRIPTION
**Pull request by jiajiechen**
@kevinthesun, I think the shape problem mentioned in https://github.com/dmlc/nnvm/issues/373 is already fixed, so I just added the `FTVMCompute`.

### CODE REVIEW

**Comment by tqchen** [comment]
@kevinthesun can you review?

**Comment by kevinthesun** [comment]
Currently matmul on;y supports 2D inputs, it would be better if we can support N-D operation, similar to numpy.

**Comment by tqchen** [comment]
It is a reasonable proposal, but if this PR is a bugfix, then we should merge this in for now. @kevinthesun can you confirm that?

File: tests/python/compiler/test_top_level1.py [hunk 1]
```diff
@@ -308,6 +313,40 @@ def forward(x):
...
+    rhs = sym.Variable('rhs')
+    result = sym.matmul(lhs, rhs, transpose_a=False, transpose_b=True)
+
+    dtype = "float32"
+    inputs = {
+        'lhs': ((1, 3), lhs),
+        'rhs': ((4, 3), rhs),
+    }
+
+    helper(result, inputs, dtype, forward, transpose_b=True)
```
Comment (kevinthesun) [inline tests/python/compiler/test_top_level1.py]: Add test for backend?

**Comment by tqchen** [comment]
error in testcase http://mode-gpu.cs.washington.edu:8080/blue/organizations/jenkins/dmlc%2Fnnvm/detail/PR-396/8/pipeline/

**Comment by tqchen** [comment]
need rebase against the master

**Comment by tqchen** [comment]
@jiajiechen please rebase against the master, let us merge this in after the rebase

**Comment by tqchen** [comment]
close because of stable issue

**Comment by wangshangsam** [comment]
Hi,

Is this issue fixed? As of today pulling from dmlc/tvm/nnvm, I'm still having the "Attribute FTVMCompute has not been registered for Operator matmul" issue, and this change doesn't seem to be merged in.

Thanks!

**Comment by srkreddy1238** [comment]
not fixed, we don't have compute for matmul now. 
I had same issue and I used transpose+dense (with out bias) for 2D.

I will PR matmul compute with 2D support using dense. This can be used until we have topi support for higher order multiply.

**Comment by wangshangsam** [comment]
Hi Siva,

Thanks for your response! The background of me having this issue is that, when you try to get the gradient graph for dense, it has matmul. Although I could somehow hack into the gradient graph and replace matmul with transpose + dense, this is not what I'm supposed to do from the perspective of the user of the tool. I would say that at current point even a 2D matmul is far more helpful than none since it's needed for gradients of a bunch of other operators.

Thanks!
Shang

**Pull request closed without merging**